pkg/controller/job/job_controller_test.go

```diff
@@ -1144,17 +1144,6 @@ func TestControllerSyncJob(t *testing.T) {
 			expectedReady:           ptr.To[int32](0),
 			expectedTerminating:     ptr.To[int32](0),
 		},
-		"indexed job with podIndexLabel feature disabled": {
```

Comment:
With emulated KEP we probably should keep this around.

Reply:
with locking the feature gate here: https://github.com/kubernetes/kubernetes/pull/128387/files#diff-530f68e01a51f788bd0ad558bde87a4a378bd3e40e3811d097029f6aa109c7a7R562, the test was failing when trying to turn it off: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128387/pull-kubernetes-unit/1850903022834552832

Reply:
Yea. You may need to change the test to use emulated version feature gate setting rather than just removing it.

Reply:
good catch @kannon92. I have updated the PR to add emulated version when testing for disabling the feature. I have added 1.31 as the EV as this will be the latest release for which the feature gate could be turned off before removal in 1.35

Reply:
Folks, in the past when graduating to GA we would remove the `if` statements from the code, for example: https://github.com/kubernetes/kubernetes/pull/126102.

I'm not sure if this policy is changed by the KEP - possibly but since the initial comment says "probably", and the KEP is quite new, let me consult with the authors: @liggitt @jpbetz and reviewer @deads2k.

Reply:
> I'm raising this because maintaining 2 branches of the code for extra 3 releases adds overhead.

@mimowo It seems that the feature gate guide was updated recently:

> For example, if a gate was `LockToDefault: true` in kubernetes version `X`, it may
be removed in version `X+3` (which must be compatible with `X+2`, `X+1`, and
`X`, all of which were also locked).

https://github.com/kubernetes/community/blame/c2a33242e1019add6e48308617f07fc6d6a35454/contributors/devel/sig-architecture/feature-gates.md#L191

Reply:
cc @Jefftree

Reply:
> Typically for full coverage, unit and integration tests exist for both when a
feature is enabled and disabled. When a feature is promoted to GA, it is usually
locked to true by default and cannot be unset in testing. For integration with
compatibility version, feature disablement tests should be maintained until the
GA feature is fully removed three releases after promotion. To test scenarios
where the feature gate may still be disabled, emulation version should be set in
disablement tests. For these disablement tests, simply set emulation version to
the version before the GA promotion to support disablement. The emulation
version would be set the line before the feature gate is set.

I think I found complete guidance here https://github.com/kubernetes/community/pull/8087, the expectation is to keep the disablement tests for the latest version before GA until the FG is removed.